packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -84,6 +84,7 @@ enum PluginRequestHeaders {
   QueryGroupID = 'X-Query-Group-Id', // mainly useful to find related queries with query splitting
   FromExpression = 'X-Grafana-From-Expr', // used by datasources to identify expression queries
   SkipQueryCache = 'X-Cache-Skip', // used by datasources to skip the query cache
+  OriginApp = 'X-Origin-App', // what application run the query
```

Comment:
Since this is for core features, the header name might be a bit misleading. How about `X-Request-Origin`?